.github/workflows/docker-ci.yml

```diff
@@ -0,0 +1,628 @@
+            fi
+            exit 1
+          else
+            echo "All required variables and secrets are set."
+          fi
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2  # Ensure enough history is available
+
+      - name: Get changed files
+        id: changed_files
+        run: |
+          files=$(git diff --name-only ${{ github.event.before }} ${{ github.sha }} '*.dockerfile')
+          echo "Files changed: $files"
+          echo "files=$files" 
+          echo "files<<EOF" >> $GITHUB_OUTPUT
+          echo "$files" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+
+
+      - name: Debug output of changed files
+        run: |
+          echo "Files from output: ${{ steps.changed_files.outputs.files }}"      
+
+
+      - name: Set proceed flag
+        id: set_proceed_flag
+        run: |
+          changed_files="${{ steps.changed_files.outputs.files }}"
+          
+          # Count files of modified Dockerfile 
+          file_count=$(echo "$changed_files" | wc -l)
+          if [ "$file_count" -eq 0 ]; then
+            echo "No Dockerfile has been modified. Setting proceed_valid to false."
+            echo "proceed_valid=false" >> $GITHUB_OUTPUT
+            exit 0
+          elif [ "$file_count" -gt 1 ]; then
+            echo "Multiple Dockerfiles have been modified ($file_count files). Setting proceed_valid to false."
+            echo "proceed_valid=false" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+
+          # Only ONE Dockerfile modified
+          file="$changed_files"
+          echo "Single Dockerfile modified: $file"
+
+          # check Dockerfile includes org.opencontainers.image.compilation=auto
+          if grep -q -E '^[^#]*LABEL\s+org\.opencontainers\.image\.compilation\s*=\s*auto' "$file"; then
+            echo "Dockerfile contains org.opencontainers.image.compilation=auto. Setting proceed_valid to true."
+            echo "proceed_valid=true" >> $GITHUB_OUTPUT
+          else
+            echo "Dockerfile does not contain org.opencontainers.image.compilation=auto. Setting proceed_valid to false."
+            echo "proceed_valid=false" >> $GITHUB_OUTPUT
+          fi
+          
+          # check Dockerfile includes org.opencontainers.image.devmode=true
+          if grep -q -E '^[^#]*LABEL\s+org\.opencontainers\.image\.devmode\s*=\s*true' "$file"; then
+            echo "Dockerfile contains org.opencontainers.image.devmode=true. Setting devmode to true."
+            echo "devmode=true" >> $GITHUB_OUTPUT
+          else
+            echo "Dockerfile does not contain org.opencontainers.image.devmode=true. Setting devmode to false."
+            echo "devmode=false" >> $GITHUB_OUTPUT
+          fi
+          
+           # check Dockerfile includes org.opencontainers.image.noscan=true
+          if grep -q -E '^[^#]*LABEL\s+org\.opencontainers\.image\.noscan\s*=\s*true' "$file"; then
```

Comment:
The `noscan` functionality might be useful but my preference would be that we add extra logic and try to capture why there is `noscan`
```yaml
if grep -q -E '^[^#]*LABEL\s+org\.opencontainers\.image\.noscan\s*=\s*true' "$file"; then
   if grep -q -E '^[^#]*LABEL\s+org\.opencontainers\.image\.noscanreason\s*' "$file"; then 
            echo "Dockerfile contains org.opencontainers.image.noscan=true. Setting trivy to noscan as a reason is provided."
            echo "noscan=true" >> $GITHUB_OUTPUT
            reason=$(grep -q -E '^[^#]*LABEL\s+org\.opencontainers\.image\.noscanreason\s*' "$file")
            echo "noscanreason=${reason}" >> $GITHUB_OUTPUT
  else
           echo "Dockerfile does not contain reason for no scan, please set org.opencontainers.image.noscanreason."
          echo "Setting proceed_valid to false."
          echo "proceed_valid=false" >> $GITHUB_OUTPUT
  fi
else 
...
fi
```
What do you think?

Reply:
Excellent advice. Patched into submission.